docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc

```diff
@@ -0,0 +1,469 @@
+-- `range`: the range of bytes you will get, for "get-by-range" operations (we
+   recommend you do simple GETs first, and then look at this);
+-- `reqUids`: the request unique ID used for logging;
+-- `callback`: your function's callback (should handle errors);
+
+=== `/lib/data/external/{{backendName}}_lib/`
+
+- this is where you should put all utility functions for your GET operation, and
+  then import then in `/lib/data/external/{{BackendName}}Client.js`, to keep
+  your code clean;
+
+=== `tests/functional/aws-node-sdk/test/multipleBackend/get/get{{BackendName}}js`
+
+- every contribution should come with thorough functional tests, showing
+  nominal context gives expected behaviour, and error cases are handled in a way
+  that is standard with the backend (including error messages and code);
+- the ideal setup is if you simulate your backend locally, so as not to be
+  subjected to network flakiness in the CI; however, we know there might not be
+  mockups available for every client; if that is the case of your backend, you
+  may test against the "real" endpoint of your data backend;
+
+=== `tests/functional/aws-node-sdk/test/multipleBackend/utils.js`
+
+NOTE: You should need this section if you have followed the tutorial in order
+      (that is, if you have covered the PUT operation already)
+
+- where you'll define a constant for your backend location matching your
+  `/tests/locationConfig/locationConfigTests.json`
+  <<location-config-test-json,test location name>>;
+- depending on your backend, the sample `keys[]` and associated made up objects
+  may not work for you (if your backend's key format is different, for example);
+  if that is the case, you should add a custom `utils.get{{BackendName}}keys()`
+
+== Operation of type DELETE
+
+DELETE routes are easy to test after PUT routes are implemented, and they are
+similar to GET routes in our implementation, hence why we're covering them
+third.
+
+These are the files you'll need to edit:
+
+=== `/lib/data/external/{{BackendName}}Client.js`
+
+- the function that is going to call your `delete()` function is also called
+  `delete()`, and it's defined in `/lib/data/multipleBackendGateway.js`;
+- define a function with signature like
+  `delete(objectGetInfo, reqUids, callback)`; this is worth exploring a
+  bit more as these parameters are the same for all backends:
+//TODO: generate this from jsdoc
+-- `objectGetInfo`: a dictionary with two entries: `key`, the object key in the
+   data store, and `client`, the data store name;
+-- `reqUids`: the request unique ID used for logging;
+-- `callback`: your function's callback (should handle errors);
+
+=== `/lib/data/external/{{backendName}}_lib/`
+
+- this is where you should put all utility functions for your DELETE operation,
+  and then import then in `/lib/data/external/{{BackendName}}Client.js`, to keep
+  your code clean;
+
+=== `tests/functional/aws-node-sdk/test/multipleBackend/get/get{{BackendName}}js`
```

Comment:
URL should be '/delete/delete{{BackendName}}.js'

Reply:
fixed thanks